docs/docs/dev_docs/getting_started/core-concepts.md

```diff
@@ -0,0 +1,90 @@
+2. Private functions are executed in the PXE, which is client-side
+3. They are rolled up and sent to the Public VM
+4. Public functions are executed in the Public VM
+5. The Public VM rolls up the private & public transaction rollups
+6. These rollups are submitted to Ethereum
+
+## Composability between private and public state
+
+The PXE is unaware of the Public VM. And the Public VM is unaware of the PXE. They are completely separate execution environments. This means:
+
+* The PXE and the Public VM cannot directly communicate with each other
+* Private transactions in the PXE are executed first, followed by public transactions
+
+You can call a public function from a private function by using this:
+
+#include_code call_public_function yarn-project/noir-contracts/src/contracts/card_game_contract/src/main.nr rust
+
+You cannot call a private function from a public function, but you can use a
+slow updates tree to read and write to public state from a private function. 
+
+### Data types
+
+Private state works with UTXOs, or what we call notes. To keep things private, everything is stored in an append-only UTXO tree, and a nullifier is created when notes are spent.
+
+Public state works similarly to other chains like Ethereum, behaving more like a public ledger. 
+
+Working with private state is like creating commitments and nullifiers, whereas working with public state is like directly updating state.
+
+We have abstractions for working with private state so you don't have to worry about these commitments and nullifiers. However, it is important to understand that the types and libraries you use will be different when working with private state and public state.
+
+For example, let's say you're trying to work with an integer. We have a library called `EasyPrivateUint` that acts like an int but in the background is actually updating notes in private state. For the public side, we instead have something called `SafeU120`. You cannot use EasyPrivateUint in a public environment, and you cannot use SafeU120 in a private environment.
+
+# Storage
+
+Currently, when writing Aztec.nr smart contracts, you will need to define two things when initiating storage:
+
+1. The storage struct, ie what you are storing and their types
+2. A storage `impl` block with `init` function that will be called when you use the storage in a function
+
+The storage struct looks like this:
+
+#include_code storage_struct yarn-project/noir-contracts/src/contracts/token_contract/src/main.nr rust
+
+The storage impl block looks like this:
+
+#include_code storage_init yarn-project/noir-contracts/src/contracts/token_contract/src/main.nr rust
+
+The `init` function must declare the storage struct with an instantiation defining how variables are accessed and manipulated. Each variable must be given a storage slot, which can be anything except 0.
+
+The impl block is likely to be abstracted away at a later date.
+
```

Comment:
Should we a link to [this](https://docs.aztec.network/dev_docs/contracts/syntax/storage/main) page for further reading?